File: languages/java/src/main/java/com/bloock/sdk/entity/identity/VerificationReceipt.java
package com.bloock.sdk.entity.identity;

import com.bloock.sdk.bridge.proto.BloockIdentityEntities;

/**
 * Represents a receipt for a verification session.
 */
public class VerificationReceipt {
  private final long sessionID;
  private final String verificationRequest;

  /**
   * Constructs a VerificationReceipt object with the specified parameters.
   * 
   * @param sessionID
   * @param verificationRequest
   */
  public VerificationReceipt(long sessionID, String verificationRequest) {
    this.sessionID = sessionID;
    this.verificationRequest = verificationRequest;
  }

  public static VerificationReceipt fromProto(BloockIdentityEntities.VerificationReceipt res) {
    return new VerificationReceipt(res.getSessionId(), res.getVerificationRequest());
  }

  public BloockIdentityEntities.VerificationReceipt toProto() {
    return BloockIdentityEntities.VerificationReceipt.newBuilder()
        .setSessionId(this.sessionID)
        .setVerificationRequest(this.verificationRequest)
        .build();
  }

  /**
   * Gets the session id of the verification.
   * 
   * @return
   */
  public long getSessionID() {
    return sessionID;
  }

  /**
   * Gets the verification request json.
   * 
   * @return
   */
  public String getVerificationRequest() {
    return verificationRequest;
  }
}

File: languages/go/entity/identity/verification_receipt.go
package identity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// VerificationReceipt represents a receipt for a verification session.
type VerificationReceipt struct {
	SessionID           int64
	VerificationRequest string
}

func NewVerificationReceiptFromProto(s *proto.VerificationReceipt) VerificationReceipt {
	if s == nil {
		return VerificationReceipt{}
	}
	return VerificationReceipt{
		SessionID:           s.SessionId,
		VerificationRequest: s.VerificationRequest,
	}
}

func (i VerificationReceipt) ToProto() *proto.VerificationReceipt {
	return &proto.VerificationReceipt{
		SessionId:           i.SessionID,
		VerificationRequest: i.VerificationRequest,
	}
}
